pkg/config/parameters.go

```diff
@@ -489,6 +490,9 @@ type Parameters struct {
 	// which strips duplicate slashes from request URL paths.
 	DisableMergeSlashes bool `yaml:"disableMergeSlashes,omitempty"`
 
+	// DisableServerHeaderTransformation signifies Envoy will not modify the Server header.
+	DisableServerHeaderTransformation http.HttpConnectionManager_ServerHeaderTransformation `yaml:"disableServerHeaderTransformation,omitempty"`
```

Comment:
Please also structure this field similarly to [DNSLookupFamily](https://github.com/projectcontour/contour/blob/5e3b212a990ee7d52a744291e19ad4cd97af6e43/pkg/config/parameters.go#L79-L95)